test/set.js

```diff
@@ -22,6 +22,22 @@ describe('set', function() {
     assert.deepEqual(b, {w: 1, x: {a: 5, b: 2, c: 3}, z: 4});
   });
 
+  it('handles empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({}, a);
+    assert.deepEqual(b, a);
+  });
+
+  it('handles nested empty object', function() {
+    var a = {w: 1, x: {a: 1, b: 2, c: 3}, z: 4};
+    var b = R.set({w: 0, x: {}, z: 3}, a);
+    assert.deepEqual(b, {w: 0, x: {a: 1, b: 2, c: 3}, z: 3});
+  });
+
```

Comment:
From that other discussion, @paldepind, you said,

> I see this function as a more powerful and useful version of assoc and deepAssoc.

I like that notion better than this change implies.

```js
    set({x: {}, z: 100}, {w: 1, x: {a: 2, b: 3}, y: 4}); 
    //=> {w: 1, x: {a: 2, b: 3}, y, 4, z: 100}

    R.pipe(R.assoc('x', {}), R.assoc('z', 100))({w: 1, x: {a: 2, b: 3}, y: 4});
    //=> {"w": 1, "x": {}, "y": 4, "z": 100}
```

I would simply expect that for flat first parameters, `set` would be consistent with something like this:

```js
    var set = R.useWith(
        R.flip(R.reduce(R.flip(R.call))), 
        R.pipe(R.toPairs, R.map(R.apply(R.assoc))), 
        R.identity
    );
```

(not that I would use such an implementation directly, but something with that behavior, namely, doing the equivalent of calling `assoc` on each value.)

But this would definitely conflict with the notion that you can't remove properties / override an object with anything other than an object.

I think we still have a significant API issue here.